src/wp-content/themes/twentytwentyone/classes/class-twenty-twenty-one-dark-mode.php

```diff
@@ -19,6 +19,11 @@ class Twenty_Twenty_One_Dark_Mode {
 	 */
 	public function __construct() {
 
+		// Disable dark-mode in the widgets editor.
+		if ( is_admin() && $_SERVER['REQUEST_URI'] && strpos( $_SERVER['REQUEST_URI'], '/widgets.php' ) ) {
```

Comment:
Is there any reason why `get_current_screen()` cannot be used here? Something like this would be a bit better, I think.

```
if ( is_admin() { 
	$screen = get_current_screen();

	if ( 'widgets' === $screen->base ) {
		return;
	}
}
```

Reply:
I tried that prior to using `$_SERVER` because I agree, it would be better... Unfortunately though it doesn't quite work in this case. `get_current_screen` here returns `null`, and even if I move it inside each function (as opposed to having it in the `__construct()`) it still doesn't function as it should.
Using `$_SERVER` is sub-optimal, but the only way I could find to make this work at this stage. Ideally we'b be able to identify the location of the editor, but right now that not possible and a larger refactor would be necessary. :disappointed:

Reply:
It looks like the widget screen does get set correctly. I've determined this by activating Query Monitor and viewing the screen tab when on the Widgets screen. The issue seems to be with when the Dark Mode class is instantiated in the theme.

Right now, `Twenty_Twenty_One_Dark_Mode` is [instantiated immediately after being included](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-content/themes/twentytwentyone/functions.php#L552-L554). This currently works, but some parts of WordPress may not be available yet.

It looks like `init` and `admin_init` are still too early to use `get_current_screen()` (this is [set up right after](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/admin.php#L165-L212) `admin_init`). I have a separate PR coming that experiments with a different way to setup the Dark Mode class.

Reply:
@aristath can you give #1486 a try when you have a chance? I _think_ this may fix the issue without having to use `$_SERVER`.

Reply:
#1486 works brilliantly in the widgets screen and is the right solution.
However, it doesn't fix the issue in the customizer... In the customizer we needed dark-mode to work properly in the preview pane but skipped in the sidebar. Is there a way to make it work there as well? (`$_SERVER['REQUEST_URI']` accounts for that since the widgets are retrieved using a request like that)